src/ast/ast.h

```diff
@@ -106,9 +106,10 @@ class Integer : public Expression {
 public:
   DEFINE_ACCEPT
 
-  explicit Integer(int64_t n, location loc);
+  explicit Integer(int64_t n, location loc, bool is_negative = true);
```

Comment:
naming nit: is_signed instead of is_negative? naively I'd think you could just check n < 0 for is_negative.

Reply:
> naively I'd think you could just check n < 0 for is_negative

The 'is_negative', which matches the parser name, is the key to understanding if we are dealing with an actual negative number or a really large unsigned integer e.g. `10223372036854775807` already comes in as casted to an `int64_t` and would show up as negative.

Reply:
I guess I don't find it that bad because we're still at the AST level which interacts with the parser which has specific logic for MINUS. We could overload Integer to accept a uint64 in the case there is no MINUS but we still need to pass this information some way to the semantic_analyser which ultimately is deciding the type.